presentation/presentation.slide

```diff
@@ -0,0 +1,254 @@
+                                               |     |   Tasks    |
+                                               +-----+            |
+      -                                               |            |
+      message with             +------------+        +----+-------+
+      task type        +-----> | Queue      |             |
+      and params               |            |             |
+                               +---------+--+             ^
+                                         |        +-------+------+
+                                         |        |              |
+                                         +------> |   Workers    |
+                                                  |              |
+                                                  +--------------+
+
+
+
+* Transactions (Async)
+
+- A transaction represents a single unit of work in centrifuge OS
+  eg: 1. Creating a document
+      2. Updating a document
+      3. Minting an NFT
+- A transaction must be considered a failure if any of the actions in it fails.
+
+* TX Manager
+
+.code ../transactions/manager.go /Manager/,/^}/
+
+* Ethereum interactions
+
+- ?
+
+* P2P
+
+- P2P layer of go-centrifuge implements
+.link https://godoc.org/github.com/centrifuge/go-centrifuge/p2p Centrifuge Wire Protocol
+- Implemented using
+.link https://libp2p.io/ Libp2p
+- Every Account in the node has the libp2p keypair used by the node in their Ethereum identity
+- Centrifuge ID concatenated to the protocol identifier is used for multiplexing requests to each Account
+
+* API
+
+- API created based on,
+.link github.com/grpc-ecosystem/grpc-gateway GRPC Gateway project
+- All the API endpoints and their requests/responses are defined as GRPC services and protobuf messages
+- Example,
+.code ../protobufs/document/service.proto /DocumentService/,/CreateDocumentProofForVersionRequest/
+
+* API docs
+- API documentation is generated with
+.link https://swagger.io Swagger Tool
+- The docs are built on CI and uploaded to
+.link https://app.swaggerhub.com/apis/centrifuge.io/cent-node centrifuge-api-docs
+
+* Storage
+
+- Any storage implementation for data store needs to comply with,
+.code ../storage/repository.go /Repository/,/^}/
+
+- Current implementation is leveldb based.
+
+* Documents
+
+A document in Centrifuge OS is represented by a Model in go-centrifuge
+
+.code ../documents/model.go /Model/,/^}/
+
+* Model Rules
+
+For ease of use and encapsulation models are constrained by following rules
+
+- A model must not have any runtime dependencies such as database. Using a compile time dep such as a lib is fine
+- A model must wrap all coredocument operations
+- No direct access to internal fields must be allowed
+
+* Document Services
+
+- A Service must be defined to handle all business logic related to a single Model.
+- Some service functions are generic enough for them to be in the `DefaultService`.
+
+* Service Interface
```

Comment:
Validators can be added as a slide to this presentation :)

Reply:
my additional slide ideas:
- ethereum testnet/ centrifuge testnet
- multi-account
- command line interface

Reply:
@manuelpolzhofer I believe what you mean by `command line interface` is how it works technically? I will make a note of that.
Regarding `multi-account` there is already a slide here `Accounts` for that, what would you like it to be covered?